### PR TITLE
Update persistentstorage dependency

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Updating transmitter to utilize the latest updates from persistent storage package

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 1]
```diff
@@ -125,14 +120,19 @@ public async ValueTask TransmitFromStorage(long maxFilesToTransmit, bool async,
 
                         if (result == ExportResult.Success)
                         {
-                            blob.Delete();
+                            blob.TryDelete();
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:123]: Only if this returned `true` should the next log a success?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs [hunk 2]
```diff
@@ -8,14 +8,12 @@
 using System.Threading;
 
 using Azure.Core.Pipeline;
+using Xunit;
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs:11]: Curious, what's the ordering convention?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs [hunk 3]
```diff
@@ -117,8 +115,9 @@ public void FailureResponseCode206()
 
             //Assert
             Assert.Single(transmitter._storage.GetBlobs());
-
-            var failedData = System.Text.Encoding.UTF8.GetString(transmitter._storage.GetBlob().Read());
+            transmitter._storage.TryGetBlob(out var blob);
+            blob.TryRead(out var content);
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs:119]: Need an assert for non-null?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:123]: No - success should be logged based on result.
If the delete fails for some reason, the same record could be transmitted twice.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 4]
```diff
@@ -23,7 +24,7 @@ namespace Azure.Monitor.OpenTelemetry.Exporter
     internal class AzureMonitorTransmitter : ITransmitter
     {
         private readonly ApplicationInsightsRestClient _applicationInsightsRestClient;
-        internal IPersistentStorage _storage;
+        internal PersistentBlobProvider _storage;
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:27]: nit: Should we consider renaming the variable? 
It makes difficult to read the code as class is provider and variable is storage.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 5]
```diff
@@ -108,15 +109,9 @@ public async ValueTask TransmitFromStorage(long maxFilesToTransmit, bool async,
                 try
                 {
                     // TODO: Do we need more lease time?
-                    var blob = _storage.GetBlob()?.Lease(10000);
-                    if (blob == null)
+                    if (_storage.TryGetBlob(out var blob) && blob.TryLease(1000))
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:112]: What if `blob` value is null?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:112]: Then tryget will return false - https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/00ac3601a6f85ed44840641a0d40c8f122732e79/src/OpenTelemetry.Extensions.PersistentStorage/FileBlobProvider.cs#L137

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs [hunk 6]
```diff
@@ -29,7 +29,7 @@ internal AzureMonitorLogExporter(ITransmitter transmitter)
             _instrumentationKey = transmitter.InstrumentationKey;
             _resourceParser = new ResourceParser();
 
-            if (transmitter is AzureMonitorTransmitter _azureMonitorTransmitter && _azureMonitorTransmitter._storage != null)
+            if (transmitter is AzureMonitorTransmitter _azureMonitorTransmitter && _azureMonitorTransmitter._fileBlobProvider != null)
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs:32]: nit: why does `_azureMonitorTransmitter` have the underscore?
I would expect that for private/internal fields, but not a local variable.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:123]: Swapping the lines, should help read the code easily.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs (hunk 2 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs]: I am not sure if there is one but general convention I see is to list 1) system 2) everything else 3) test. Updated the prdering

Comment (reyang) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/OfflineStorageTests.cs]: Roger that, thanks!

**Review by reyang — APPROVED** [review]

**Pull request merged**